### ISSUE TITLE
[Bug]: Page Drag&Drop/Floating exception

### ISSUE THREAD

**Issue by iomismo** [Krypton-Suite/Standard-Toolkit#653]
**Describe the bug**
Exception when dragging a page inside a DockableWorkspace.

**Only happens in 60.22.2.32.**

Stack trace
```
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height, PixelFormat format)
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height)
   at Krypton.Toolkit.CommonHelper.ScaleImageForSizedDisplay(Image src, Single trgtWidth, Single trgtHeight)
   at Krypton.Toolkit.ButtonSpecView.GetImage(PaletteState state)
   at Krypton.Toolkit.RenderStandard.AllocateImageSpace(StandardContentMemento memento, IPaletteContent paletteContent, IContentValues contentValues, PaletteState state, Rectangle displayRect, RightToLeft rtl, Size[,]& allocation)
   at Krypton.Toolkit.RenderStandard.GetContentPreferredSize(ViewLayoutContext context, IPaletteContent palette, IContentValues values, VisualOrientation orientation, PaletteState state, Boolean composition, Boolean glowing)
   at Krypton.Toolkit.ViewDrawContent.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewLayoutDocker.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewComposite.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewDrawSplitCanvas.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewDrawButton.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewComposite.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewLayoutCenter.GetPreferredSize(ViewLayoutContext context)
   at Krypton.Toolkit.ViewDrawDocker.Layout(ViewLayoutContext context)
   at Krypton.Toolkit.ViewDecorator.Layout(ViewLayoutContext context)
   at Krypton.Toolkit.ViewDrawDocker.Layout(ViewLayoutContext context)
   at Krypton.Toolkit.ViewManager.Layout(ViewLayoutContext context)
   at Krypton.Toolkit.KryptonForm.CheckViewLayout()
   at Krypton.Toolkit.KryptonForm.WindowChromePaint(Graphics g, Rectangle bounds)
   at Krypton.Toolkit.VisualForm.OnNonClientPaint(IntPtr hWnd)
   at Krypton.Toolkit.VisualForm.OnWM_NCPAINT(Message& m)
   at Krypton.Toolkit.VisualForm.WndProc(Message& m)
   at Krypton.Docking.KryptonFloatingWindow.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)
```

Add a Form
Add KryptonPanel
Add DockableWorkspace to the panel

Setup docking manager on Form Load:
```csharp
KryptonDockingWorkspace w = KryptonDockingManager.ManageWorkspace(kryptonDockableWorkspace);
KryptonDockingManager.ManageControl(kryptonPanel, w);
KryptonDockingManager.ManageFloating(this);
```
Add some page using KryptonDockingManager.AddToWorkspace()

In this minimal application it only throws exception when trying to drag a page. On another more complex application it also throws the same exception at form resize or even when trying to show the desktop.

**Comment by iomismo** [comment]
Uploaded test app
[https://github.com/iomismo/KryptonDockableWorkspaceTest](https://github.com/iomismo/KryptonDockableWorkspaceTest)

The problem is in Docking Customized demo, which uses DockableWorkspace.

System.ArgumentException: "Parameter is not valid"

**Comment by thiagoraheem** [comment]
The property currentHeight is -1 when dragging out the form

**Comment by thiagoraheem** [comment]
I put this code and then it is working for me

![image](https://user-images.githubusercontent.com/7240990/157495182-f512ecc0-f4ec-48cc-946f-8d49b595a87b.png)

**Comment by Wagnerp** [comment]
Hi @iomismo please re-test using `nightly` build >= .94

**Comment by Wagnerp** [comment]
Hi @iomismo is the fix working?

**Comment by iomismo** [comment]
Hi @Wagnerp it is working, many thanks. Although I directly used last production nugget version.

BUT, using, for example, Krypton.Docking.DockingEdge.Left in version > 6.2201.4, seems that it doesn't render pages titles and their widths are "truncated".

E.g.:
```
DockingManager.AddAutoHiddenGroup("Control", Krypton.Docking.DockingEdge.Left, new KryptonPage[] { page1, page2, page3 });
```
After pinning a page (or simply no autohiding pages):

**6.2201.4**
![pinned_6_2201_4](https://user-images.githubusercontent.com/5861929/162573864-704c86ae-ca94-4e34-832f-6b1184e9763e.png)

**65.22.4.94**
![Captura de pantalla 2022-04-09 141213](https://user-images.githubusercontent.com/5861929/162573885-7ddc1dfb-e9a5-4540-bd6d-cf23e60b4af1.png)

I know this is another issue, but, dont know why, I expected it to be resolved by the fixed floating issue, and I totally forgot to mention it.

Thanks!

**Comment by bakerhillpins** [comment]
@iomismo

Can confirm that issue. Simply change the package version that is referenced on the "Multi Control Docking 2019" (or any) example to see the sizing change.  Took me over a day of messing around to figure out that it was simply the package version I was using compared to the samples.

Samples reference Version="6.2110.274-beta" which I can't even find but it apparently exist somewhere in NuGet. The code doesn't have any tags and I don't have time to sort through dates of release to figure it out ATM. 

I guess I'll drop down to the version where it works. Thanks for posting that BTW! Documentation is basically nonexistent (unless I'm missing something) so it's hard to understand how to manage sizing anyway.

**Comment by Wagnerp** [comment]
Hi @bakerhillpins 

> Samples reference Version="6.2110.274-beta" which I can't even find but it apparently exist somewhere in NuGet.

All packages post `6.22.x` have been unlisted due to version conflicts 2110 > 22. Documentation is available here: https://krypton-suite.github.io/Standard-Toolkit-Online-Help/Source/Help/Output/index.html, though a downloadable version can be obtained here: https://github.com/Krypton-Suite/Help-Files/releases

Hope this helps!

**Comment by Wagnerp** [comment]
Hi @iomismo 

Great to hear! Can you just confirm that this happens in all versions, or just in the latest stable?

Thanks

**Comment by bakerhillpins** [comment]
@Wagnerp  

Yes, that does help. I had issues navigating the webpage earlier.  I appear to be actually getting pages to load today so I I'm just going to file that under /PEBKAC. I've downloaded the .exe from that link and will take a look at that shortly. 

Ah.. the [Api Documentation](https://krypton-suite.github.io/Standard-Toolkit-Online-Help/Source/Help/Output/api/index.html) is where I was the other day and it was empty but I suppose that's in the code. I also don't see anything that talks about using the other packages besides the Toolkit, such as Docking, which is where I'm struggling ATM. I'll check the download.  Apologies for sounding like I'm complaining, I'm just trying to find this stuff without using up anyone's else's time.

No need to confuse this issue with anymore noise. I'll just keep digging in the code for now. Thanks for your efforts with this!

**Comment by Wagnerp** [comment]
@bakerhillpins 

No problem, I do need to update the downloads anyway, since it's still using the February release, but it should still be ok. Post a feature request to remind me to put a link to the EXEs in the readme file.

**Comment by bakerhillpins** [comment]
Done!  https://github.com/Krypton-Suite/Standard-Toolkit/issues/691

The help.exe has all the API docs I'm looking for at the moment!  Thanks!

**Comment by iomismo** [comment]
Hi @Wagnerp

> Great to hear! Can you just confirm that this happens in all versions, or just in the latest stable?

It happens in stable versions > 6.2201.4 (60.x and 65.x)

**Comment by Smurf-IV** [comment]
Solved with  #699
